docs/language/learn-ql/cpp/basic-query-cpp.rst

```diff
@@ -0,0 +1,145 @@
+|                                                               |                                                                                                                   |                                                                                                                        |
+|                                                               | ``select`` statements for queries that are used to find instances of poor coding practice are always in the form: |                                                                                                                        |
+|                                                               | ``select <program element>, "<alert message>"``                                                                   |                                                                                                                        |
++---------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+
+Extend the query
+----------------
+
+Query writing is an inherently iterative process. You write a simple query and then, when you run it, you discover examples that you had not previously considered, or opportunities for improvement.
+
+Remove false positive results
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Browsing the results of our basic query shows that it could be improved. Among the results you are likely to find examples of ``if`` statements with an ``else`` branch, where an empty ``then`` branch does serve a purpose. For example:
+
+.. code-block:: cpp
+
+   if (...) {
+     ...
+   } else if (!strcmp(option, "-verbose") {
+     // nothing to do - handled earlier
+   } else {
+     error("unrecognized option");
+   }
+
+In this case, identifying the ``if`` statement with the empty ``then`` branch as redundant is a false positive. One solution to this is to modify the query to ignore empty ``then`` branches if the ``if`` statement has an ``else`` branch.
+
+To exclude ``if`` statements that have an ``else`` branch:
+
+#. Extend the ``where`` clause to include the following extra condition:
+
+   .. code-block:: ql
+
+      and not ifstmt.hasElse()
+
+   The ``where`` clause is now:
+
+   .. code-block:: ql
+
+      where ifstmt.getThen() = block and
+        block.getNumStmt() = 0 and
+        not ifstmt.hasElse()
+
+#. Click **Run**.
+
+   There are now fewer results because ``if`` statements with an ``else`` branch are no longer reported.
+
+➤ `See this in the query console <https://lgtm.com/query/1505895837823/>`__
```

Comment:
The query uses `not exists(ifstmt.getElse())` instead of the previously described `not ifstmt.hasElse()`.

Reply:
Good point! Updated.